src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleDictionary.cs

```diff
@@ -167,11 +170,11 @@ public static partial class JsonSerializer
 
             if (value == null)
             {
-                writer.WriteNull(key);
+                writer.WriteNull(key!);
```

Comment:
Why do we need `!` here and below? We already asserted that `key != null` and if it is, we throw above (line 165).

Reply:
I think it is left over from update for netfx compiler not understanding Debug.Assert()